File: SerenityExamples.Web/Modules/ServerTypes/Tables/OrderLinesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderLinesRow } from "./OrderLinesRow";

export interface OrderLinesColumns {
    OrderNumber: Column<OrderLinesRow>;
    ItemNumber: Column<OrderLinesRow>;
    ProductNumber: Column<OrderLinesRow>;
    Qty: Column<OrderLinesRow>;
    Price: Column<OrderLinesRow>;
    Total: Column<OrderLinesRow>;
}

export class OrderLinesColumns extends ColumnsBase<OrderLinesRow> {
    static readonly columnsKey = 'Tables.OrderLines';
    static readonly Fields = fieldsProxy<OrderLinesColumns>();
}

File: SerenityExamples.Web/Modules/Tables/OrderLines/OrderLinesColumns.cs
namespace SerenityExamples.Tables.Columns;

[ColumnsScript("Tables.OrderLines")]
[BasedOnRow(typeof(OrderLinesRow), CheckNames = true)]
public class OrderLinesColumns
{
    [EditLink]
    public string OrderNumber { get; set; }

    [EditLink]
    public int ItemNumber { get; set; }

    [EditLink]
    public string ProductNumber { get; set; }

    [EditLink]
    public int Qty { get; set; }

    [EditLink]
    public int Price { get; set; }

    [EditLink]
    public int Total { get; set; }
}